src/emulator/functionsEmulator.ts

```diff
@@ -795,6 +861,39 @@ export class FunctionsEmulator implements EmulatorInstance {
     return true;
   }
 
+  addBlockingTrigger(url: string, blockingTrigger: BlockingTrigger): boolean {
+    logger.debug(`addBlockingTrigger`, JSON.stringify({ blockingTrigger }));
+
+    const eventType = blockingTrigger.eventType;
+    if (AUTH_BLOCKING_EVENTS.includes(eventType as any)) {
+      const accessToken = blockingTrigger.options!.accessToken as boolean;
+      const idToken = blockingTrigger.options!.idToken as boolean;
+      const refreshToken = blockingTrigger.options!.refreshToken as boolean;
+      if (blockingTrigger.eventType === BEFORE_CREATE_EVENT) {
+        this.blockingFunctionsConfig.triggers = {
+          ...this.blockingFunctionsConfig.triggers,
+          beforeCreate: {
+            functionUri: url,
+          },
+        };
+      } else {
+        this.blockingFunctionsConfig.triggers = {
+          ...this.blockingFunctionsConfig.triggers,
+          beforeSignIn: {
+            functionUri: url,
+          },
+        };
+      }
+      this.blockingFunctionsConfig.forwardInboundCredentials = {
+        accessToken,
+        idToken,
+        refreshToken,
+      };
+    }
+
+    return true;
```

Comment:
I think this function needs to return false if it fails to add the blocking trigger for any reason. This could happen if the if-condition in L868 fails for some reason.

Reply:
Good catch thanks